graylog2-rest-client/src/main/java/org/graylog2/restclient/lib/Version.java

```diff
@@ -95,6 +99,31 @@ public Version() {
         commitSha1 = null;
     }
 
+    @JsonCreator
+    public static Version fromString(String version) {
+        if (isNullOrEmpty(version))
+            return null;
+
+        final Matcher matcher = versionPattern.matcher(version);
+
+        if (matcher.matches()) {
+            final int major = Integer.parseInt(matcher.group(1));
+            final int minor = Integer.parseInt(matcher.group(2));
+            final int patch = Integer.parseInt(matcher.group(3));
+            final String additional = matcher.group(5);
+            final String sha1 = matcher.group(7);
+
+            if (isNullOrEmpty(sha1)) {
+                if (isNullOrEmpty(additional))
+                    return new Version(major, minor, patch);
+                else
+                    return new Version(major, minor, patch, additional);
+            } else
+                return new Version(major, minor, patch, additional, sha1);
+        } else
+            return null;
```

Comment:
It might make sense to return a "default" Version instance instead of `null` here (as constructed by the default constructor of this class) so that clients do not need to check for `null` values.